### PR TITLE
Http2 test for IIS

### LINKED ISSUE #14139: Kestrel and IIS set different Request.Protocol values for HTTP/2
Kestrel sets `HTTP/2` while IIS sets `HTTP/2.0`.

Is this expected behavior?

### PR DESCRIPTION
**Pull request by jkotalik**
Fixes https://github.com/aspnet/AspNetCore/issues/14139.

cc @gfoidl

### CODE REVIEW

**Review by JamesNK — APPROVED** [review]

File: src/Servers/IIS/IIS/test/testassets/InProcessWebSite/Startup.cs [hunk 1]
```diff
@@ -63,6 +63,11 @@ private async Task ServerAddresses(HttpContext ctx)
             await ctx.Response.WriteAsync(string.Join(",", serverAddresses.Addresses));
         }
 
+        private async Task CheckIsHttp2(HttpContext ctx)
```
Comment (Tratcher) [inline src/Servers/IIS/IIS/test/testassets/InProcessWebSite/Startup.cs]:
```suggestion
        private async Task CheckProtocol(HttpContext ctx)
```

File: src/Servers/IIS/IntegrationTesting.IIS/src/IISDeploymentParameterExtensions.cs [hunk 2]
```diff
@@ -37,7 +37,7 @@ public static void AddHttpsToServerConfig(this IISDeploymentParameters parameter
 
                     element.Descendants("access")
                         .Single()
-                        .SetAttributeValue("sslFlags", "Ssl, SslNegotiateCert");
+                        .SetAttributeValue("sslFlags", "None");
```
Comment (jkotalik) [inline src/Servers/IIS/IntegrationTesting.IIS/src/IISDeploymentParameterExtensions.cs]: I need to read into setting this to None. Setting this to none makes it seem like SSL is disabled for the entire site. I'm still seeing in the application that it is binding to https and using HTTP/2. Maybe it's using h2c?

**Review by Tratcher — APPROVED** [review]

File: src/Servers/IIS/IntegrationTesting.IIS/src/IISDeploymentParameterExtensions.cs (hunk 2 above)
Comment (jkotalik) [reply src/Servers/IIS/IntegrationTesting.IIS/src/IISDeploymentParameterExtensions.cs]: Nvrm, I think it's a poor doc. I'm seeing https on the server.

Comment (halter73) [reply src/Servers/IIS/IntegrationTesting.IIS/src/IISDeploymentParameterExtensions.cs]: So what does setting this to None actually do?

**Review by halter73 — APPROVED** [review]

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/HttpsTests.cs [hunk 3]
```diff
@@ -87,10 +89,26 @@ public async Task ServerAddressesIncludesBaseAddress()
...
+            var deploymentParameters = Fixture.GetBaseDeploymentParameters(HostingModel.InProcess);
+            deploymentParameters.ApplicationBaseUriHint = $"https://localhost:{port}/";
+            deploymentParameters.AddHttpsToServerConfig();
+            deploymentParameters.SetWindowsAuth(false);
+
+            var deploymentResult = await DeployAsync(deploymentParameters);
+            var client = CreateNonValidatingClient(deploymentResult);
+            client.DefaultRequestVersion = HttpVersion.Version20;
+
+            Assert.Equal("HTTP/2", await client.GetStringAsync($"/CheckIsHttp2"));
```
Comment (gfoidl) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/HttpsTests.cs]: Let's test other protocol versions too?

File: src/Servers/IIS/IntegrationTesting.IIS/src/IISDeploymentParameterExtensions.cs (hunk 2 above)
Comment (jkotalik) [reply src/Servers/IIS/IntegrationTesting.IIS/src/IISDeploymentParameterExtensions.cs]: It removes client cert validation I believe.

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/HttpsTests.cs (hunk 3 above)
Comment (jkotalik) [reply src/Servers/IIS/IIS/test/Common.FunctionalTests/HttpsTests.cs]: Other tests already verify this behavior (checking http 1.1)

**Pull request merged**